### ISSUE TITLE
IDTAXA upgraded to SILVA r138

### ISSUE THREAD

**Issue by digitalwright** [benjjneb/dada2#950]
The SILVA r138 classifier has just been released on the [DECIPHER](http://www2.decipher.codes/Downloads.html) website. Please consider upgrading the DADA2 [tutorial](https://benjjneb.github.io/dada2/tutorial.html) to reflect this version. Thanks!

**Comment by adriaaula** [comment]
Hello! 

I was checking the taxonomy of my bacteria with this new version and I had a surprising result. Is it possible that augmenting the DB lowered the taxonomic assignation? 

![image](https://user-images.githubusercontent.com/15213478/75276357-38b75680-57c3-11ea-864f-09d968660737.png)

**Comment by digitalwright** [comment]
If I understand correctly, the surprising result is that there are 1% fewer reads assigned to a taxon?  Yes, that is possible when the reference set is updated. There are fewer groups in the SILVA r138 taxonomy. For example, there are 9740 genera in r132 and 6679 genera in r138. There are about 200k fewer sequences too. My guess is that this is due to the correction of errors that is described [here](https://www.arb-silva.de/documentation/release-138/).

**Comment by adriaaula** [comment]
It is not to a taxon. I have ~6000 different taxons and the plot is the NA distribution for each one, sorry, i didn't quite explain the plot. 

Thanks for the comment, now it makes way more sense.

**Comment by WAEL1990** [comment]
Hello
I followed the the DADA2 [tutorial](https://benjjneb.github.io/dada2/tutorial.html) to carry out the assignment of the taxonomy (6400 seq) but I received this error. 

`>` ids <- IdTaxa(dna, trainingSet, strand="both", verbose=T) # use all processors

Error in getMethod(f, c("XRawList", "XRawList")) : 
  no method found for function 'match' and signature XRawList, XRawList
SVP are there any ideas to solve it.

**Comment by benjjneb** [comment]
Note to self: Update Silva to v138 for both `assignTaxonomy` and `IdTaxa` in the upcoming tutorial revision.

**Comment by juanboja** [comment]
> 
> 
> Hello
> I followed the the DADA2 [tutorial](https://benjjneb.github.io/dada2/tutorial.html) to carry out the assignment of the taxonomy (6400 seq) but I received this error.
> 
> `>` ids <- IdTaxa(dna, trainingSet, strand="both", verbose=T) # use all processors
> 
> Error in getMethod(f, c("XRawList", "XRawList")) :
> no method found for function 'match' and signature XRawList, XRawList
> SVP are there any ideas to solve it.

Hi, I am having the same problem. I can't figure out the error. I ran the code 10 days ago and it worked, now I try it and it gives me the error.. I have loaded the XVector library. Any idea what it could be?

**Comment by digitalwright** [comment]
This error looks like you haven't loaded `library(DECIPHER)`.

**Comment by juanboja** [comment]
> 
> 
> This error looks like you haven't loaded `library(DECIPHER)`.

That was my first thought, but I checked on that. I posted it into another post https://github.com/benjjneb/dada2/issues/658#issuecomment-627102491

Anyways, I don't know what was happening, but as always, a good reboot solved the problem.. It is working again. Thanks

**Comment by benjjneb** [comment]
Woops, I thought I had already done this.

**Comment by Sebastian-Mynott** [comment]
I have this issue as well. 



> Hello
> I followed the the DADA2 [tutorial](https://benjjneb.github.io/dada2/tutorial.html) to carry out the assignment of the taxonomy (6400 seq) but I received this error.
> 
> `>` ids <- IdTaxa(dna, trainingSet, strand="both", verbose=T) # use all processors
> 
> Error in getMethod(f, c("XRawList", "XRawList")) :
> no method found for function 'match' and signature XRawList, XRawList
> SVP are there any ideas to solve it.

However, in my case rebooting has not solved the issue. Any assistance would be much appreciated.

**Comment by Sebastian-Mynott** [comment]
> I have this issue as well.
> 
> > Hello
> > I followed the the DADA2 [tutorial](https://benjjneb.github.io/dada2/tutorial.html) to carry out the assignment of the taxonomy (6400 seq) but I received this error.
> > `>` ids <- IdTaxa(dna, trainingSet, strand="both", verbose=T) # use all processors
> > Error in getMethod(f, c("XRawList", "XRawList")) :
> > no method found for function 'match' and signature XRawList, XRawList
> > SVP are there any ideas to solve it.
> 
> However, in my case rebooting has not solved the issue. Any assistance would be much appreciated.

UPDATE: I managed to get DECIPHER::IdTaxa() running but only after restarting R and before installation of DADA2, so I reckon something in DADA2 is getting in the way of the IdTaxa function here.
